### ISSUE TITLE
Some textfields have hard to read text with Arc Dark

### ISSUE THREAD

**Issue by danhm** [horst3180/arc-firefox-theme#44]
Some text fields have either light text on a light background or dark text on a dark background. I've attached a few examples. I'm using Firefox 43.0.1 and Ark Dark 43.20151213 in Firefox. I'm using arc-theme 1450051815.946cbf5 in Debian Testing with Openbox.

![gog](https://cloud.githubusercontent.com/assets/10888280/11987854/957bf75c-a99b-11e5-890a-4323fe4aa955.png)
![schwab](https://cloud.githubusercontent.com/assets/10888280/11987855/957f72ba-a99b-11e5-9982-ebe84779db72.png)

**Comment by ghost** [comment]
Apparently it's a bug with Firefox. However [Numix](https://github.com/shimmerproject/Numix) has solved this problem so I think it's solvable with Arc as well.

In the meantime, I put this script into my `~/bin` folder and give it a filename of `firefox` and when I run firefox this script runs instead:

https://gist.github.com/xHN35RQ/999c4c861a9506dfd507

**Comment by danhm** [comment]
Thanks Nick! That'll work for now.

**Comment by lnalex** [comment]
Personally, I use a homemade CSS which fix dark inputs and textareas with [Stylish plugin](https://addons.mozilla.org/en-US/firefox/addon/stylish/) : 

https://gist.github.com/lnalex/5366f3b0d027eaa4940c

I'm open to any suggestion to improve it

**Comment by danhm** [comment]
Cool. You can do that by modifying
~/.mozilla/firefox/$PROFILE/chrome/userContent.css as well, no need for an
extension.

**Comment by alxlion** [comment]
Of course. I don't like to override directly the files, an update could broke everything (I don't really know for user profile).

**Comment by tostiheld** [comment]
@lnalex do you have the css from that gist somewhere? the link appears to be broken (ends in a 404)

**Comment by tostiheld** [comment]
@lnalex do you have the css from that gist somewhere? the link appears to be broken (ends in a 404)

**Comment by alxlion** [comment]
@tostiheld Yes sorry, I've deleted it inadvertently but I don't use it anymore (it was not a real fix). I'm waiting for a real fix for that issue...

**Comment by tostiheld** [comment]
so this

```
html input:not[type=button] {
  background-color: #ffffff !important;
  color: #000000 !important;
}
html textarea {
  background-color: #ffffff !important;
  color: #000000 !important;
}
```

is the 'dirty' fix you can do in stylish or userchrome (at least it works perfectly for me)

i have also tracked down the culprit with the inspector (not that hard). the stylesheet where the colors come from is resource://gre-resources/forms.css and the specific colors for the back- and foreground are -moz-Field and -moz-FieldText respectively. i tried overriding them in chome/global/in-content/common.css, but that didn't work. i tried overriding the colors directly as wel as assigning new colors to the input and textarea elements. i'm stuck now. i'm also not a css expert.

**Comment by alxlion** [comment]
According to [ArchWiki](https://wiki.archlinux.org/index.php/Firefox#Firefox_startup_takes_very_long#Unreadable_input_fields_with_dark_GTK.2B_themes) , you can fix the problem by creating a new Stylish theme:

```
input:not(.urlbar-input):not(.textbox-input):not(.form-control):not([type='checkbox']) {
    -moz-appearance: none !important;
    background-color: white;
    color: black;
}

#downloads-indicator-counter {
    color: white;
}

textarea {
    -moz-appearance: none !important;
    background-color: white;
    color: black;
}

select {
    -moz-appearance: none !important;
    background-color: white;
    color: black;
}
```

**Comment by qazip** [comment]
I'm having the same problem:

![arc_ycombinator](https://cloud.githubusercontent.com/assets/16175716/20236599/2fa7ae5a-a8b2-11e6-9a4c-c46bb5882e06.png)

and

![dark_google_bug](https://cloud.githubusercontent.com/assets/16175716/20236601/3781ec3a-a8b2-11e6-9ec7-a99717567b6e.png)

Is there going to be a non-dirty fix soon? I'm using the beta release (else the firefox' menu is broken)

Thank you for dark btw, fantastic theme.

**Comment by ghost** [comment]
Same problem here as mimf. Using Arc-Dark GTK theme with Xubuntu 16.04 and Arc-Dark in Firefox 49 makes checkboxes, radiobuttons, "resize-grip" of textarea, scrollbars, textarea itself ugly. Dark-grey background and black text. E. g. Google Calendar entries are hard to edit. The workarounds as mentioned by Inalex leave radio, checkboxes and "resize-grip" of untouched and ugly.

Changing only Firefox Theme to Arc-Darker makes bookmark-iconbar hard to read.

The combination of Firefox Arc-Darker theme and the linux commandline "env GTK_THEME=Arc-Darker firefox" to start firefox works. Unfortunately firefox only is "half-dark".

The optimum solution would be to modify Arc-Dark GTK-theme to display Firefox "content window" with Arc (the bright one) theme settings. Most webpages that don't theme controls like checkboxes or radios themselves seem to expect a light/bright gtk theme.

**Comment by teslaji** [comment]
This problem still persist for me, is there any optimum solution for it ? yet.

**Comment by fossfreedom** [comment]
is this the same issue as this? https://github.com/horst3180/arc-firefox-theme/issues/108

**Comment by alexhulbert** [comment]
Heads up: If you add an "inherit" after each css line, it fixes some css elements which are implicitly set to a color through CSS. For example, if a textbox is implied to be purple by its parent, this new css snippet won't force it to be white. Don't ask me why it works, but it does and it works well.

```
input:not(.urlbar-input):not(.textbox-input):not(.form-control):not([type='checkbox']) {
    -moz-appearance: none !important;
    background-color: white;
    background-color: inherit;
    color: black;
    color: inherit;
}

#downloads-indicator-counter {
    color: white;
    color: inherit;
}

textarea {
    -moz-appearance: none !important;
    background-color: white;
    background-color: inherit;
    color: black;
    color: inherit;
}

select {
    -moz-appearance: none !important;
    background-color: white;
    background-color: inherit;
    color: black;
    color: inherit;
}
```

**Comment by teslaji** [comment]
Where shall I make this changes !? please let me know the location of this
CSS files.

**Comment by alxlion** [comment]
@teslaji You need [Stylish addon](https://addons.mozilla.org/en-US/firefox/addon/stylish/?src=external-userstyleshome)

**Comment by francozappa** [comment]
cc: @teslaji @lnalex 

I've copy-pasted the @Taconut snippet to firefox's `your-profile-id/chrome/userContent.css` file and it worked as well.

**Comment by ghost** [comment]
There's also this addon on, which works with current Nightly (there's a short flickr when it re-paints the input field though, at least for me), which is suppose to work with other "dark" themes.

https://addons.mozilla.org/en-US/firefox/addon/text-contrast-for-dark-themes/

**Comment by SpecialBomb** [comment]
The stylish fix also still leaves some style issues in forms and iframes, so it still doesn't fix everything unfortunately. Tested this on my own website that uses buttons for forms, and they all get messed up with the fix on, and look normal with it off.

**Comment by Legogris** [comment]
Updated @Taconut 's style to not hide radio buttons:
```
input:not(.urlbar-input):not(.textbox-input):not(.form-control):not([type='checkbox']):not([type='radio']) {
    -moz-appearance: none !important;
    background-color: white;
    background-color: inherit;
    color: black;
    color: inherit;
}

#downloads-indicator-counter {
    color: white;
    color: inherit;
}

textarea {
    -moz-appearance: none !important;
    background-color: white;
    background-color: inherit;
    color: black;
    color: inherit;
}

select {
    -moz-appearance: none !important;
    background-color: white;
    background-color: inherit;
    color: black;
    color: inherit;
}
```

**Comment by 9joshua** [comment]
Using @Legogris style above. Any way to show the drop down arrow for a selection box?...
![image](https://user-images.githubusercontent.com/20234985/34861498-449336e6-f7ca-11e7-930a-d16c2d5d7960.png)

Should look like this...
![image](https://user-images.githubusercontent.com/20234985/34861571-957f37b2-f7ca-11e7-9141-10032b0c6cd5.png)

**Comment by adhuliya** [comment]
I can confirm this issue still exists with the latest of version of the arc and firefox. Still no out-of-the-box solution.

**Comment by vijay609** [comment]
I am having the same issue with dark text being rendered on dark background.

**Comment by WesR** [comment]
The issues still persist for me. I found the best fix by following this tutorial: 
https://www.mkammerer.de/blog/gtk-dark-theme-and-firefox/

Steps (All credit to Moritz Kammerer):
1. Open `about:config` in Firefox
2. Create a new String value (right click), named `widget.content.gtk-theme-override`
3. Set a light GTK theme name as the value, I chose `Arc-Darker`
4. Restart Firefox
